code/game/machinery/_machinery.dm

```diff
@@ -248,24 +248,39 @@ Class Procs:
 	return !(machine_stat & (NOPOWER|BROKEN|MAINT))
 
 /obj/machinery/can_interact(mob/user)
-	var/silicon = issiliconoradminghost(user)
-	if((machine_stat & (NOPOWER|BROKEN)) && !(interaction_flags_machine & INTERACT_MACHINE_OFFLINE))
+	var/silicon = issilicon(user)
+	var/admin_ghost = IsAdminGhost(user)
+	var/living = isliving(user)
+
+	if((machine_stat & (NOPOWER|BROKEN)) && !(interaction_flags_machine & INTERACT_MACHINE_OFFLINE)) // Check if the machine is broken, and if we can still interact with it if so
 		return FALSE
-	if(panel_open && !(interaction_flags_machine & INTERACT_MACHINE_OPEN))
+
+	if(panel_open && !(interaction_flags_machine & INTERACT_MACHINE_OPEN)) // Check if we can interact with an open panel machine, if the panel is open
 		if(!silicon || !(interaction_flags_machine & INTERACT_MACHINE_OPEN_SILICON))
 			return FALSE
 
-	if(silicon)
+	if(silicon || admin_ghost) // If we are an AI or adminghsot, make sure the machine allows silicons to interact
```

Comment:
you don't need this for aghosts, admins have a verb for toggling AI interaction, save on the proc calls chief

Reply:
IsAdminGhost() only returns true if the admin AI interact is active. This is its intended use.

Reply:
ah ok, my mistake